ssh/server/auth/publickey_test.go

```diff
@@ -0,0 +1,386 @@
+package auth
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"errors"
+	"testing"
+
+	gliderssh "github.com/gliderlabs/ssh"
+	internalclientMocks "github.com/shellhub-io/shellhub/pkg/api/internalclient/mocks"
+	"github.com/shellhub-io/shellhub/pkg/models"
+	"github.com/shellhub-io/shellhub/ssh/pkg/metadata"
+	metadataMocks "github.com/shellhub-io/shellhub/ssh/pkg/metadata/mocks"
+	"github.com/shellhub-io/shellhub/ssh/pkg/sshsrvtest"
+	"github.com/shellhub-io/shellhub/ssh/pkg/target"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	gossh "golang.org/x/crypto/ssh"
+)
+
+func generateTestPubKey(t *testing.T) gliderssh.PublicKey {
+	privateKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	publicRsaKey, err := gossh.NewPublicKey(&privateKey.PublicKey)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	return publicRsaKey
+}
+
+func TestPublicKeyHandler(t *testing.T) {
+	cases := []struct {
+		description string
+		setup       func(ctx *gliderssh.Context) *sshsrvtest.Conn
+		mocks       func(ctx gliderssh.Context)
+		expected    bool
+	}{
+		{
+			description: "fails when could not store the target",
+			setup: func(ctx *gliderssh.Context) *sshsrvtest.Conn {
+				srv := sshsrvtest.New(
+					&gliderssh.Server{
+						Handler: func(s gliderssh.Session) {
+							*ctx = s.Context()
+						},
+					},
+					&gossh.ClientConfig{
+						User: "user@namespace.00-00-00-00-00-00",
+						Auth: []gossh.AuthMethod{
+							gossh.Password("secret"),
+						},
+						HostKeyCallback: gossh.InsecureIgnoreHostKey(),
+					},
```

Comment:
How is possible to test the PublicKeyHandler using only Password on `gossh.AuthMethod`?

Reply:
The test does not test the connection itself. Instead, it passes the session's context directly to the `PublicKeyHandler()` as shown below:
```go
func TestPublicKeyHandler(t *testing.T) {
    // [...]
    for _, tc := range cases {
        t.Run(tc.description, func(t *testing.T) {
            var ctx gliderssh.Context

            srv := tc.setup(&ctx)
            defer srv.Teardown()

            srv.Start()
            assert.NoError(t, srv.Agent.Run("cmd"))

            tc.mocks(ctx)

            result := PublicKeyHandler(ctx, generateTestPubKey(t)) // <----
            assert.Equal(t, tc.expected, result) // <---
        })
    }
}
```
This approach allows us to test the handler as a unit and retrieve the result for use in `assert.Equal()`.

Reply:
It is similar to the way the `PasswordHandler()` is tested:
```go
func TestPasswordHandler(t *testing.T) {
    // [...]
    for _, tc := range cases {
        t.Run(tc.description, func(t *testing.T) {
            var ctx gliderssh.Context

            srv := tc.setup(&ctx)
            defer srv.Teardown()

            srv.Start()
            assert.NoError(t, srv.Agent.Run("cmd"))

            tc.mocks(ctx)

            result := PasswordHandler(ctx, tc.password) // <---
            assert.Equal(t, tc.expected, result) // <---
        })
    }
}
```
This consistency in testing methods helps maintain a standardized approach across different handlers, making the testing process more organized.